client/svsm.go

```diff
@@ -0,0 +1,93 @@
+package client
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/google/go-configfs-tsm/configfs/linuxtsm"
+	"github.com/google/go-configfs-tsm/report"
+	"github.com/google/go-tpm/tpm2/transport"
+	"github.com/google/go-tpm/tpm2"
+	sabi "github.com/google/go-sev-guest/abi"
+	sg "github.com/google/go-sev-guest/client"
+	sevpb "github.com/google/go-sev-guest/proto/sevsnp"
+	rpb "github.com/google/go-tpm-tools/proto/register_credential"
+	tpb "github.com/google/go-tpm-tools/proto/tpm"
+)
+
+// MakeSVSMAttestation creates a SevSnpSvsmAttestation containing all the information needed to verify the SVSM e-vTPM.
+func MakeSVSMAttestation(tpm transport.TPM, req *tpb.ImportBlob, keyAlgo tpm2.TPMAlgID, teeNonce [sabi.ReportDataSize]byte) (*rpb.SevSnpSvsmAttestation, error) {
+	certified, err := SolveChallengeImportCertify(tpm, req, keyAlgo)
+	if err != nil {
+		return nil, fmt.Errorf("SolveChallengeImportCertify() = %w", err)
```

Comment:
here and elsewhere, please match the error string style. for example: "failed to create import certify solution".